### ISSUE TITLE
Better factorization algorithms

### ISSUE THREAD

**Issue by risubaba** [sympy/sympy#18341]
Is it in the scope of sympy to have better factorisation algorithms such as quadratic sieve which work well for numbers having as many as 100 digits?

**Comment by jksuom** [comment]
Yes, it would be useful. See https://github.com/sympy/sympy/issues/12409 and https://github.com/sympy/sympy/pull/12436.

**Comment by abhinav28071999** [comment]
I am working on implementing quadratic sieve, it's a long code. Right now it's around half done. Will send a PR in a few days.
